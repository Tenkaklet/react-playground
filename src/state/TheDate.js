import React from 'react';

class TheDate extends React.Component {

    constructor (props) {
        super(props);
        this.state = { datetime: new Date() };
        console.log('constructor');
    }

    componentDidMount() {
        console.log('ComponentDidMount');
        this.interval  = setInterval(() => {
            this.setState({
                datetime: new Date()
            })
        }, 1000);
    }

    componentWillMount() {
        clearInterval(this.interval);
    }


    render() {
        return (
            <div>
                {this.state.datetime.toLocaleDateString()}
            </div>
        )
    }
}

export default TheDate;