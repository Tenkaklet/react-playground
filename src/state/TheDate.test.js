import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TheDate from './TheDate';

describe('The Date Component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TheDate />, div);

    it.skip('Renders the UI as expected', () => {
        expect(renderer.create(<TheDate />).toJSON()).toMatchSnapshot();
    });
});