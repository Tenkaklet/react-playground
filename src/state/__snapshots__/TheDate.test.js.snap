// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Date Component Renders the UI as expected 1`] = `
<div>
  2/18/2019
</div>
`;
